### ISSUE TITLE
Error in terminal following what seems to be a successful linting operation

### ISSUE THREAD

**Issue by afholderman** [sasstools/sass-lint#1057]
I run my sass-lint command and get a list of errors but at the end get an error, details below.

**What version of Sass Lint are you using?** 
1.10.2

**Please include any relevant parts of your configuration**
sass-lint.yml

```
# Linter Options
options:
  # Don't merge default rules
  merge-default-rules: false
  max-warnings: 0
# File Options
files:
  include: 'stye/scss/**/*.s+(a|c)ss'
  ignore:
    - 'style/scss/variables.scss'
    - 'node_modules/**/*'

```

**What did you do? Please include the actual source code causing the issue.** 
Ran `sass-lint -config .sass-lint.yml './style/scss/**/*.scss' -verbose -no-exit`

**What did you expect to happen?** 
Linting with no error

**What actually happened? Please include any error messages given to you by Sass Lint.**
Got the following error in the terminal:
```
/Users/aholderman/Sites/devdesktop/manifest-dev/docroot/themes/manifest/node_modules/sass-lint/index.js:313
    throw new exceptions.SassLintFailureError(errorCount.count + ' errors were detected in \n- ' + errorCount.files.join('\n- '));
    ^
Error
    at Function.sassLint.failOnError (/Users/aholderman/Sites/devdesktop/manifest-dev/docroot/themes/manifest/node_modules/sass-lint/index.js:313:11)
    at detectPattern (/Users/aholderman/Sites/devdesktop/manifest-dev/docroot/themes/manifest/node_modules/sass-lint/bin/sass-lint.js:31:10)
    at /Users/aholderman/Sites/devdesktop/manifest-dev/docroot/themes/manifest/node_modules/sass-lint/bin/sass-lint.js:83:5
    at Array.forEach (native)
    at Object.<anonymous> (/Users/aholderman/Sites/devdesktop/manifest-dev/docroot/themes/manifest/node_modules/sass-lint/bin/sass-lint.js:82:16)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
```

**Comment by tnguyen14** [comment]
I have the same problem.

**Comment by DanPurdy** [comment]
What version of node are you both using? I'd like to try and replicate this as closely as possible. Thanks

**Comment by tnguyen14** [comment]
I am on node `v6.9.4`

**Comment by tnguyen14** [comment]
I see this problem even with a minimal .sass-lint.yml file with just 2 errors.

**Comment by tnguyen14** [comment]
It seems to be due to this line https://github.com/sasstools/sass-lint/blob/master/bin/sass-lint.js#L30
Though I can't tell what `program.exit` is.

**Comment by afholderman** [comment]
I'm running node 7.8.0.

**Comment by Belelros** [comment]
Same issue here

**Comment by DanielSundberg** [comment]
I get this as well, any ideas?

**Comment by maxpolski** [comment]
Just to many errors or warnings due to

```
if (lint.errorCount(detects).count || tooManyWarnings(detects, userConfig)) {
  exitCode = 1;
}
```
line 16 in sass-lint.js

**Comment by Amadiere** [comment]
Removing `max-warnings` did not resolve the issue for me.
To confirm, the file is being generated successfully and it appears to be OK - but outputs the error message in the OP.

**Comment by DanPurdy** [comment]
I will come back and check this but the command you're using is only using single dahses before the keywords. It should be 2 `--verbose` not `-verbose` and `--no-exit` instead of `-no-exit` you can use single dashes with the shorthands. `-v` & `-q`. As i say this is just me guessing at first look but will go and confirm it soon.

**Comment by rbf** [comment]
First of all, thanks for maintaining and caring about the useful `sass-lint` tool! Linting for the win! 🎉 

I have the same issue reported here. It looks like `sass-lint` (v1.11.1) is throwing an error in the cases when it should simple return with a non-zero exit code (i.e. when some errors were found or when the number of warning exceeds the `--max-warnings` value).

The errors are consistently `SassLintFailureError` ([thrown here](https://github.com/sasstools/sass-lint/blob/v1.11.1/index.js#L313)) and `MaxWarningsExceededError` ([thrown here](https://github.com/sasstools/sass-lint/blob/v1.11.1/index.js#L317)) matching the assumption above.

Maybe the line [`lint.failOnError(detects, configOptions, configPath);`](https://github.com/sasstools/sass-lint/blob/v1.11.1/bin/sass-lint.js#L31) (mentioned by @tnguyen14 above) should catch those two exceptions and cleanly exit with `1`?

I hope this helps understanding the issue.

For the record, my current environment:

```
$ node --version
v8.5.0
$ sass-lint --version
1.11.1
$ cat package.json 
{
  "name": "my-project",
  "private": true,
  "dependencies": {
    "@rails/webpacker": "^3.0.1",
    "babel-preset-react": "^6.24.1",
    "prop-types": "^15.5.10",
    "react": "^15.6.1",
    "react-dom": "^15.6.1",
    "react-on-rails": "^9.0.2"
  },
  "devDependencies": {
    "babel-eslint": "^8.0.0",
    "eslint": "^3.19.0 || ^4.3.0",
    "eslint-config-airbnb": "^15.1.0",
    "eslint-plugin-css-modules": "^2.7.5",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-jsx-a11y": "^5.1.1",
    "eslint-plugin-react": "^7.1.0",
    "sass-lint": "^1.11.1",
    "webpack-dev-server": "^2.8.1"
  }
}
```

**Comment by DanPurdy** [comment]
Thanks @rbf I will look into this for sure. we're going to do a little overhaul of a few things shortly and these errors / Cli are on the list.

**Comment by markgoho-EDT** [comment]
Still getting this error @DanPurdy - any updates on this?

**Comment by E314c** [comment]
this is fixed by using the following code around [this line](https://github.com/sasstools/sass-lint/blob/master/bin/sass-lint.js#L31)
```javascript
  if (program.exit) {
    try{
      lint.failOnError(detects, configOptions, configPath);
    }catch (e){
      console.error(e.message);
    }
    process.exit(exitCode);
  }
```
I would submit a pull request, but i'm not going to jump through all the hoops mentioned in the `contributing` file.
Anyone may take this code and use it, put it in this project, or just go into your own `node_modules/sass-lint/bin/sass-lint.js` and make the corrections.
If you're wondering why you're not seeing the actually lint errors, apparently: this module is silent by default, and you must supply the `--verbose` flag in the CLI if you want to see what went wrong.
(Which I think is pretty stupid).

Also, how does something like this take 7 months to implement? I diagnosed in ~20mins and coded it in 2?

**Comment by filidorwiese** [comment]
I've made a PR to fix this issue. In the meantime I also noticed that when using the `--no-exit` flag, the exception is not thrown to stdout. Could be a workaround if you need it.

**Comment by mdesantis** [comment]
Hey there! Any updates about this issue? It's pretty annoying and a patch has already been submitted, so maybe it could be merged :-)
